ommx_fixstars_amplify_adapter/adapter.py

```diff
@@ -0,0 +1,289 @@
+import amplify
+
+from typing import Union
+
+from ommx.v1 import (
+    Solution,
+    Instance,
+    DecisionVariable,
+    Constraint,
+    State,
+)
+from ommx.v1.function_pb2 import Function
+from ommx.adapter import SolverAdapter
+from ommx.v1.linear_pb2 import Linear
+from ommx.v1.quadratic_pb2 import Quadratic
+from ommx.v1.polynomial_pb2 import Polynomial
+from ommx.v1.constraint_pb2 import Constraint as RawConstraint
+from ommx.v1.decision_variables_pb2 import DecisionVariable as RawDVar
+
+from .exception import OMMXFixstarsAmplifyAdapterError
+
+
+class OMMXFixstarsAmplifyAdapter(SolverAdapter):
+    def __init__(self, ommx_instance: Instance):
+        """
+        :param ommx_instance: The ommx.v1.Instance to solve.
+        """
+        self.instance = ommx_instance
+        self.model = amplify.Model()
+
+        self._set_decision_variables()
+        self._set_objective()
+        self._set_constraints()
+
+    @staticmethod
+    def solve(
+        ommx_instance: Instance, amplify_token: str = "", timeout: int = 1000
+    ) -> Solution:
+        """Solve the given ommx.v1.Instance using Fixstars Amplify, returning an
+        ommx.v1.Solution.
+
+        **NOTE** The `amplify_token` parameter _must_ be passed to properly
+          instantiate the Fixstars Client. Using the default value will result
+          in an error.
+
+        :param ommx_instance: The ommx.v1.Instance to solve.
+        :param amplify_token: Token for instantiating the Fixstars Client, obtained from your Fixstars Amplify account.
+        :param timeout: Timeout passed the client
+
+        Example:
+        =========
+        The following example shows how to solve an unconstrained linear optimization problem with `x` as the objective function.
+
+        .. doctest::
+
+            >>> from ommx_fixstars_amplify_adapter import OMMXFixstarsAmplifyAdapter
+            >>> from ommx.v1 import Instance, DecisionVariable, Linear
+            >>>
+            >>> x1 = DecisionVariable.integer(1, lower=0, upper=5)
+            >>> ommx_instance = Instance.from_components(
+            ...     decision_variables=[x1],
+            ...     objective=x1,
+            ...     constraints=[],
+            ...     sense=Instance.MINIMIZE,
+            ... )
+            >>> token = "YOUR API TOKEN" # Set your API token
+            >>> solution = OMMXFixstarsAmplifyAdapter.solve(ommx_instance, amplify_token=token) # doctest: +SKIP
+        """
+        if amplify_token == "":
+            raise OMMXFixstarsAmplifyAdapterError(
+                "No Fixstars Amplify token specificed -- cannot instantiate client"
+            )
+
+        adapter = OMMXFixstarsAmplifyAdapter(ommx_instance)
+
+        client = amplify.FixstarsClient()
+        client.token = amplify_token
+        client.parameters.timeout = timeout
+
+        result = amplify.solve(adapter.model, client)
```

Comment:
```suggestion
        result = amplify.solve(adapter.solver_input, client)
```
It is better to use `adapter.solver_input` instead of `adapter.model`. Because using `adapter.model` directly is not intended with [the guide](https://jij-inc.github.io/ommx/ja/user_guide/solver_adapter.html).